src/components/DisplayMap.js

```diff
@@ -0,0 +1,145 @@
+import  { useEffect, useRef } from "react";
+import { makeStyles } from "@material-ui/core/styles";
+import "ol/ol.css";
+import { Map, View } from "ol";
+import { Tile as TileLayer, Vector as VectorLayer } from "ol/layer";
+import { OSM, Vector as VectorSource } from "ol/source";
+import { Style, Icon } from "ol/style";
+import Feature from "ol/Feature";
+import Point from "ol/geom/Point";
+import LocationPin from "../images/marker-icon.png";
+import Select from "ol/interaction/Select";
+import Overlay from "ol/Overlay";
+import * as ReactDOM from "react-dom";
+// import './DisplayMap.css';
+
+const useStyles = makeStyles((theme) => ({
+  popupContainer: {
+    width: "100%",
+    height: "600px",
+    position: "relative",
+  },
+  popupContent: {
+    backgroundColor: "white",
+    border: "1px solid #ccc",
+    padding: "10px",
+    boxShadow: "0px 2px 6px rgba(0, 0, 0, 0.1)",
+    maxWidth: "300px",
+  },
+  popupContentStrong: {
+    display: "block",
+    fontWeight: "bold",
+    marginBottom: "8px",
+  },
+  olPopup: {
+    maxWidth: "none",
+  },
+}));
+
+const DisplayMap = ({ tasks }) => {
```

Comment:
Rename to `Map`

Reply:
To make this component more generic, pass in geoJSON features as props.  
Then you can make a wrapper function that creates these features from tasks